iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -79,9 +79,7 @@ private NetworkXml() {
     private static Set<String> getNetworkExtensions(Network n) {
         Set<String> extensions = new TreeSet<>();
         for (Identifiable<?> identifiable : n.getIdentifiables()) {
-            for (Extension<? extends Identifiable<?>> extension : identifiable.getExtensions()) {
-                extensions.add(extension.getName());
-            }
+            identifiable.getExtensions().stream().filter(e -> !e.isEmpty()).forEach(e -> extensions.add(e.getName()));
```

Comment:
I would use the serializer to check if an extension should be serialized or not.

Reply:
ok as said above